lib/vfs/src/passthru_fs.rs

```diff
@@ -0,0 +1,89 @@
+//! `PassthruFileSystem` simply forwards and "owns" a filesystem,
+//! passing the operations to the inner filesystem
+
+use std::path::Path;
+#[allow(unused_imports, dead_code)]
+use tracing::{debug, error, info, trace, warn};
+
+use crate::*;
+
+#[derive(Debug)]
+pub struct PassthruFileSystem {
```

Comment:
I think the use-case is that you don't need to write `Box<dyn FileSystem + Send + 'static>` everywhere you use it.

Reply:
Sure, but why not just `type DynSharedFilesystem = Arc<dyn Filesystemt + Send + 'static>` ?

Reply:
And have an `impl Filesystem for Arc<dyn Filesystem + Send + 'static>`.

Reply:
This is what John said:

- PassthruFS: wraps a boxed file system with an implemented trait VirtualSystem - this is needed so that a Box<dyn VirtualFileSystem> can be wrapped in an Arc and shared - some of the interfaces pass around a Box<dyn VirtualFileSystem>

I don't have any more info than this.

Reply:
@john-sharratt , maybe you can provide some context here.

Reply:
I still think this is pointless, but we can remove this in a follow up.